File: Python_Samples/connect_customer_to_agent.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

from pprint import pprint
import requests

from settings import sid, token


def connect_customer_to_agent(sid, token,
                              agent_no, customer_no, callerid,
                              timelimit=None, timeout=None, calltype='trans'):
    return requests.post('https://twilix.exotel.in/v1/Accounts/{sid}/Calls/connect.json'.format(sid=sid),
        auth=(sid, token),
        data= {
            'From': agent_no,
            'To': customer_no,
            'CallerId': callerid,
            'TimeLimit': timelimit,
            'TimeOut': timeout,
            'CallType': calltype
        })


if __name__ == '__main__':
    r = connect_customer_to_agent(
        sid, token,
        agent_no="<First-phone-number-to-call (Your agent's number)>",
        customer_no="<Second-phone-number-to-call (Your customer's number)>",
        callerid="<Your-Exotel-virtual-number>",
        timelimit="<time-in-seconds>",  # This is optional
        timeout="<time-in-seconds>",  # This is also optional
        calltype="trans"  # Can be "trans" for transactional and "promo" for promotional content
        )
    print r.status_code
    pprint(r.json())

File: PHP_Samples/connect_customer_to_agent.php
<?php

$post_data = array(
    'From' => "<First-phone-number-to-call (Your agent's number)>",
    'To' => "<Second-phone-number-to-call (Your customer's number)>",
    'CallerId' => "<Your-Exotel-virtual-number>",
    'TimeLimit' => "<time-in-seconds> (optional)",
    'TimeOut' => "<time-in-seconds (optional)>",
    'CallType' => "promo" //Can be "trans" for transactional and "promo" for promotional content
);
 
// You can get your $exotel_sid, $api_key and $api_token from: https://my.exotel.com/apisettings/site#api-credentials 
$api_key = "xxxxx"; // Your `API KEY`.
$api_token = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"; // Your `API TOKEN`
$exotel_sid = "xxxxxxxxxxx" // Your `Account Sid`
 
$url = "https://".$api_key.":".$api_token."@twilix.exotel.in/v1/Accounts/".$exotel_sid."/Calls/connect";
 
$ch = curl_init();
curl_setopt($ch, CURLOPT_VERBOSE, 1);
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_FAILONERROR, 0);
curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($post_data));
 
$http_result = curl_exec($ch);
$error = curl_error($ch);
$http_code = curl_getinfo($ch ,CURLINFO_HTTP_CODE);
 
curl_close($ch);
 
print "Response = ".print_r($http_result);

?>
